crates/transaction-pool/src/error.rs

```diff
@@ -288,4 +288,15 @@ impl InvalidPoolTransactionError {
             }
         }
     }
+
+    /// Returns `true` if an import failed due to nonce gap.
+    pub const fn is_nonce_gap(&self) -> bool {
```

Comment:
do we need this?

Reply:
yes I will use this to address https://github.com/paradigmxyz/reth/issues/7144